Analysis client pluging panics when gossip is disabled

## Bug description

When started with the gossip plugin  (and thus also autopeering selection) disabled, the analysis client crashes, because `autopeering.Selection` is nil.

## Steps To reproduce the bug

Start GoShimmer from the `develop` branch with `--node.disablePlugins=gossip`.

## Errors

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xabe30a]

goroutine 90 [running]:
github.com/iotaledger/hive.go/autopeering/selection.(*Protocol).GetOutgoingNeighbors(...)
	/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200403132600-4c10556e08a0/autopeering/selection/protocol.go:105
github.com/iotaledger/goshimmer/plugins/analysis/client.reportHeartbeat(0xc0007220a8)
	/goshimmer/plugins/analysis/client/plugin.go:92 +0x5a
github.com/iotaledger/goshimmer/plugins/analysis/client.Run.func1(0xc00071c240)
	/goshimmer/plugins/analysis/client/plugin.go:43 +0x3a1
github.com/iotaledger/hive.go/daemon.(*OrderedDaemon).runBackgroundWorker.func1(0x11fc548, 0xc00009f600, 0xc0001a8100)
	/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200403132600-4c10556e08a0/daemon/daemon.go:115 +0x34
created by github.com/iotaledger/hive.go/daemon.(*OrderedDaemon).runBackgroundWorker
	/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200403132600-4c10556e08a0/daemon/daemon.go:114 +0xe7
```

--- a/plugins/analysis/client/plugin.go
+++ b/plugins/analysis/client/plugin.go
@@ -88,20 +88,26 @@ func reportHeartbeat(dispatchers *EventDispatchers) {
 		nodeID = local.GetInstance().ID().Bytes()
 	}
 
-	// Get outboundIds (chosen neighbors)
-	outgoingNeighbors := autopeering.Selection.GetOutgoingNeighbors()
-	outboundIds := make([][]byte, len(outgoingNeighbors))
-	for i, neighbor := range outgoingNeighbors {
-		// Doesn't copy the ID, take care not to modify underlying bytearray!
-		outboundIds[i] = neighbor.ID().Bytes()
-	}
+	var outboundIds [][]byte
+	var inboundIds [][]byte
 
-	// Get inboundIds (accepted neighbors)
-	incomingNeighbors := autopeering.Selection.GetIncomingNeighbors()
-	inboundIds := make([][]byte, len(incomingNeighbors))
-	for i, neighbor := range incomingNeighbors {
-		// Doesn't copy the ID, take care not to modify underlying bytearray!
-		inboundIds[i] = neighbor.ID().Bytes()
+	// When gossip (and autopeering selection) is enabled, we have neighbors to report
+	if autopeering.Selection != nil {
+		// Get outboundIds (chosen neighbors)
+		outgoingNeighbors := autopeering.Selection.GetOutgoingNeighbors()
+		outboundIds = make([][]byte, len(outgoingNeighbors))
+		for i, neighbor := range outgoingNeighbors {
+			// Doesn't copy the ID, take care not to modify underlying bytearray!
+			outboundIds[i] = neighbor.ID().Bytes()
+		}
+
+		// Get inboundIds (accepted neighbors)
+		incomingNeighbors := autopeering.Selection.GetIncomingNeighbors()
+		inboundIds = make([][]byte, len(incomingNeighbors))
+		for i, neighbor := range incomingNeighbors {
+			// Doesn't copy the ID, take care not to modify underlying bytearray!
+			inboundIds[i] = neighbor.ID().Bytes()
+		}
 	}
 
 	packet := &heartbeat.Packet{OwnID: nodeID, OutboundIDs: outboundIds, InboundIDs: inboundIds}

